test/perf/run_nightly_perf.py

```
@@ -554,40 +559,44 @@ def collect_profiles(ns, epp_pod_name, results_dir, test_name, run_time_clean, p
-            print("Profile collector: timed out waiting for log line 'Stage 1 - run started'")
-            break
-            
-        if log_proc.poll() is not None:
-            break
-            
-        line = log_proc.stdout.readline()
-        if not line:
-            time.sleep(1)
-            continue
+    while time.time() - start_time < timeout:
+        log_proc = subprocess.Popen(
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)